### PR TITLE
Test tictactoe

### PR DESCRIPTION
**Pull request by dasileker**
### Testing Ruby with RSpec - Tic Tac Toe
-In this milestone, we used RSpec to create a testing file for our Tic Tac Toe project.

### CODE REVIEW

File: spec/tic_tac_toe_spec.rb [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+    end
+
+    describe '#position_taken?' do
+      it 'returns true/false based on whether the position on the board is already occupied' do
+        game = TicTacToe.new
+        board = ['X', ' ', ' ', ' ', ' ', ' ', ' ', ' ', 'O']
+        game.instance_variable_set(:@board, board)
+
+        index = 0
+        expect(game.position_taken?(index)).to be(true)
```
Comment (expjazz) [inline spec/tic_tac_toe_spec.rb]: - [x] You should not expect multiple conditions from one _**it**_ block. Test all the positive and negative cases on different _**it**_ blocks. (This repeats sometimes in your code. Kindly fix them)
image

File: spec/tic_tac_toe_spec.rb [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        allow_any_instance_of(TicTacToe).to receive(:position_taken?).and_return(false)
+        game = TicTacToe.new
+        expect(game.valid_move?(99)).to be_falsey
+      end
+    end
+
+    describe '#turn_count' do
+      it 'counts occupied positions' do
+        game = TicTacToe.new
+        board = ['O', ' ', ' ', ' ', 'X', ' ', ' ', ' ', 'X']
```
Comment (expjazz) [inline spec/tic_tac_toe_spec.rb]: - [x] You could use the let and define the variables that you use more than one time on the top of the file (this **_board_**, for instance, you use on line 106 as well). You also have other repetitive variables. Re-check all of them and aply the let for those you use more than once.

**Review by expjazz — CHANGES_REQUESTED** [review]
## Status: Changes Required :recycle: 

Hey team :+1: 

You did a great job on this project. You covered a lot of different scenarios :1st_place_medal: 

However, we have a couple of problems:

- [x] You have some repetition in your variables. You should use the keyword let and define them at the top of your files.

- [x] In some cases, you are testing multiple conditions for one **_it_** block. Each **_it_** block should expect one condition only

For more detail, kindly check the inline comments below.

### Please make the required changes and resubmit for a new code review once you are ready
Good Luck and Happy Coding :pray:

_P.S. If you have any questions about my review or need help, you can message me on slack (@ Expedito Andrade).:smile:_

File: spec/tic_tac_toe_spec.rb (hunk 1 above)
Comment (dasileker) [reply spec/tic_tac_toe_spec.rb]: done

File: spec/tic_tac_toe_spec.rb (hunk 2 above)
Comment (dasileker) [reply spec/tic_tac_toe_spec.rb]: done I did refactor codes now its shorter

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello Team! 👋 
Awesome job so far!! 👍

sadly, the test cases are both showing errors,

image

I doubled checked the recent commits to make sure the problem wasn't on my end, and they worked fine.

### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Comment by SpaYco** [comment]
Tips: 
- use `require_relative` instead of `require` 
- when you use `require`, the tests would run from where your terminal is currently is, using `require _relative` would use the current file's directory. so keep an eye on that!

fixing the issues with those will make your tests work again!

**Review by peterrobert — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED :recycle: :stop_sign: 
### HELLO :wave: YOU HAVE DONE AN IMPRESSIVE JOB ON THIS PROJECT. YOU HAVE TESTED ALL YOUR METHODS :clap: HOWEVER, THERE IS STILL SOME CHANGES REQUIRED :arrow_down: 

- [x] You have created all the positive tests. However, there are no negative tests for example:
in your Check win? it returns true if the player wins :+1: You also need to test the other expectation example what if the player loses? it's actually a requirement :arrow_down: 
commonsense

### if you have a question concerning the review kindly reach me on slack @peterrobert

### HAPPY CODING :computer:

**Review by maelfosso — APPROVED** [review]
# Congratulations :clap: 

Well done job :smiley_cat: 
You can merge it now :+1: 

Good continuation :handshake: 

Mael FOSSO

**Pull request merged**